pkg/controllers/management/auth/crtb_handler.go

```diff
@@ -101,65 +101,66 @@ type crtbLifecycle struct {
 }
 
 func (c *crtbLifecycle) Create(obj *v3.ClusterRoleTemplateBinding) (runtime.Object, error) {
-	obj, err := c.reconcileSubject(obj)
+	var localConditions []metav1.Condition
+	obj, err := c.reconcileSubject(obj, &localConditions)
 	return obj, errors.Join(err,
-		c.reconcileBindings(obj),
-		c.updateStatus(obj))
+		c.reconcileBindings(obj, &localConditions),
+		c.updateStatus(obj, localConditions))
 }
 
 func (c *crtbLifecycle) Updated(obj *v3.ClusterRoleTemplateBinding) (runtime.Object, error) {
-	obj.Status.LocalConditions = []metav1.Condition{}
-	obj, err := c.reconcileSubject(obj)
+	var localConditions []metav1.Condition
+	obj, err := c.reconcileSubject(obj, &localConditions)
 	return obj, errors.Join(err,
-		c.reconcileLabels(obj),
-		c.reconcileBindings(obj),
-		c.updateStatus(obj))
+		c.reconcileLabels(obj, &localConditions),
+		c.reconcileBindings(obj, &localConditions),
+		c.updateStatus(obj, localConditions))
 }
 
 func (c *crtbLifecycle) Remove(obj *v3.ClusterRoleTemplateBinding) (runtime.Object, error) {
 	condition := metav1.Condition{Type: clusterRoleTemplateBindingDelete}
 
 	if err := c.mgr.reconcileClusterMembershipBindingForDelete("", pkgrbac.GetRTBLabel(obj.ObjectMeta)); err != nil {
 		c.s.AddCondition(&obj.Status.LocalConditions, condition, failedToDeleteClusterMembershipBinding, err)
-		return nil, errors.Join(err, c.updateStatus(obj))
+		return nil, errors.Join(err, c.updateStatus(obj, obj.Status.LocalConditions))
```

Comment:
No local slice in Remove ?

Reply:
Remove is kind of special. We'll just add a new condition if there is an error when deleting. No condition is added if the resource is successfully deleted. 

For create or updates we build all the conditions by watching the current status in the cluster, and just update the status if conditions are different. We don't do this when removing. That's why I think is better to use `.Status.LocalConditions` so we don't loose the other conditions.